aea/helpers/async_utils.py

```diff
@@ -543,7 +543,7 @@ async def _run_wrapper(self) -> None:
         """Wrap run() method."""
         if not self._completed_event or not self._loop:  # pragma: nocover
             raise ValueError("Start was not called!")
-
+        self._is_running = True
```

Comment:
This was missing, revert and run reverted associated test to see warning @marcofavorito

Reply:
Could set `_is_running=False` after line 551 @solarw ?